.changeset/curly-squids-chew.md

```diff
@@ -0,0 +1,17 @@
+---
+"@chakra-ui/pin-input": minor
+---
+
+- Move input props logic to parent hook and expose prop-getter `getInputProps`.
+  This helps to co-locate the state logic for easier debugging
+- Added support for alpha-numeric and secret values.
+- Added `type` prop which can be either `alphanumeric` or `number`
+- Added `mask` prop to hide input value similar to `<input type='password' />`
+
+```jsx
+<PinInput type="alphanumeric" mask>
+  <PinInputField />
+  <PinInputField />
+  <PinInputField />
+</PinInput>
+```
```

Comment:
Nice! I love putting an example of usage directly in the changelog! Makes it much easier to digest the change than trying to parse through all of the above text and understand what it means.